### PR TITLE
Fix duplicate tail of StaticFiles with index_file.

### PR DESCRIPTION
**Pull request by garyhai**
And fix the unit test cases by add request path. 

Move the PR from 'master' to '0.6 branch.
#323

### CODE REVIEW

**Comment by garyhai** [comment]
'0.6' branch is stable for integrated test. HTTPS and pipeline cannot work in 'master' branch. HTTPS failure is by cause of crate h2 version 0.1.10.

File: src/fs.rs [hunk 1]
```diff
@@ -653,10 +653,6 @@ impl<S: 'static> Handler<S> for StaticFiles<S> {
                     // TODO: It'd be nice if there were a good usable URL manipulation
                     // library
                     let mut new_path: String = req.path().to_owned();
-                    for el in relpath.iter() {
```
Comment (DoumanAsh) [inline src/fs.rs:656]: eh just to be sure I understand this code here correctly:
We put double request path into redirect path?

**Review by DoumanAsh — APPROVED** [review]
Looks good, thanks, though I guess we'll need to merge it on master

**Comment by DoumanAsh** [comment]
@fafhrd91 should we merge this change to both `0.6` and master?

**Comment by fafhrd91** [comment]
@DoumanAsh could you take care of this. I am on UB refactoring.

**Comment by DoumanAsh** [comment]
Ok, I'll merge it to both branches after work

File: src/fs.rs (hunk 1 above)
Comment (garyhai) [reply src/fs.rs:656]: Yes. When 'index.html' file in subfolder within static folder,  access the URL to the path of the subfolder will reproduce this error by double request path in field LOCATION of HTTP.  For the static_index example of (https://github.com/actix/examples.git), when we move the file index.html to subfolder "static_index/static/test", 'curl -i localhost:8080/test' got the response below.

``` http
HTTP/1.1 302 Found
content-length: 0
location: /testtest/index.html
date: Mon, 25 Jun 2018 10:11:34 GMT
```

**Comment by DoumanAsh** [comment]
Merged on master too https://github.com/actix/actix-web/commit/a9425a866bdb7d2be8cd2ab0ab9232541f93258a

**Pull request merged**